src/MSBuild/CommandLineSwitches.cs

```diff
@@ -50,7 +50,8 @@ internal enum ParameterlessSwitch
 #if DEBUG
             WaitForDebugger,
 #endif
-            NumberOfParameterlessSwitches
+            NumberOfParameterlessSwitches,
```

Comment:
If I underastand ti correctly `NumberOfParameterizedSwitches` has to be last as it is used for countof(ParameterlessSwitch).
Please move `FeautureAvailability` above it and add comment to `NumberOfParameterlessSwitches` with meaning "This has be be kept as last enum value"